### PR TITLE
Removing dependency on embedded OctopusCLI

### PR DESCRIPTION
**Pull request by johnsimons**
The purpose of this PR is to reduce the need for the following dependencies:

1. ship an embedded version of Octopus CLI
2. use a "portable" version of the Octopus CLI

Going forward, pipelines should include the `Octopus CLI Installer` task. Customers should not assume availability of the embedded version included in this plugin. This change requires maintenance by users for the benefit of more control; to specify the version of the Octopus CLI. It also removes the dependency to include .NET Core, which has caused problems for customers if/when dependencies change.

This PR includes a new version of the `Octopus CLI Installer` task that downloads a self-contained version of the Octopus CLI. We continue to support having the Octopus CLI pre-installed in the agent and added to the `PATH` environment variable.

We also took this opportunity to deprecate the `OctopusPack` task in favour of the [built it `ArchiveFiles`](https://docs.microsoft.com/en-au/azure/devops/pipelines/tasks/utility/archive-files?view=azure-devops) task instead.

One last change, the `OctopusCreateRelease` task now supports adding release notes in a text field directly as part of creation (this functionality was considered legacy in V4).

### CODE REVIEW

**Review by slewis74 — COMMENTED** [review]
I've had a quick scan through. Thoughts so far
- I'm not sure about removing the Pack step. If we're keeping the CLI for now, then should we remove it? I guess you could still use the V4 Pack step, right?
- Create Release, I'd like to see us drop the Deployment parts off this step, it should be just about creating a release. The CLI should be creating an output variable that can be used to get the Version number in the subsequent step, if necessary.
- Create Release, the Changeset code is legacy and should be removed. I think we should retain the free text and also the filename, I suspect there are cases where the latter is being used (e.g. the release notes are in a package or something like that). Without telemetry we can't tell, so let's leave that field there for now.

**Comment by johnsimons** [comment]
@slewis74 
> I'm not sure about removing the Pack step. If we're keeping the CLI for now, then should we remove it? I guess you could still use the V4 Pack step, right?

You can use V4, but the question is why? I thought we agreed zip is what the majority of people need? Have I missed any other edge cases?

**Comment by johnsimons** [comment]
@slewis74 
> Create Release, I'd like to see us drop the Deployment parts off this step, it should be just about creating a release. The CLI should be creating an output variable that can be used to get the Version number in the subsequent step, if necessary.

I didn't want to change too much in this release, should we raise a public issue for each change? How do we migrate users from CreateRelease + Deployment to just CreateRelease?

**Comment by johnsimons** [comment]
@slewis74 
> Create Release, the Changeset code is legacy and should be removed.

I have removed it

> I think we should retain the free text and also the filename, 

I have kept the text form, not the filename, given it was not exposed in the UI before, users can always fall back to the "additional args" for it?

File: source/tasks/BuildInformation/BuildInformationV5/task.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            "required": true,
+            "helpMarkDown": "Normally, if the same package build information already exists on the server, the server will reject the package build information push. This is a good practice as it ensures build information isn't accidentally overwritten or ignored. Use this setting to override this behavior.",
+            "options": {
+                "false": "Fail if exists",
+                "true": "Overwrite existing",
+                "IgnoreIfExists": "Ignore if exists"
+            }
+        },
+        {
+            "name": "AdditionalArguments",
```
[the remark this answers is not in the document]
Comment (johnsimons) [reply source/tasks/BuildInformation/BuildInformationV5/task.json:76]: @slewis74 looking in our [doco](https://octopus.com/docs/octopus-rest-api/octopus-cli/build-information), I find the additional args useful for things like increasing the `timeout` or changing the `logLevel`, what do you think?

File: source/tasks/CreateOctopusRelease/CreateOctopusReleaseV5/createRelease.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+export class CreateRelease {
+    constructor(readonly tool: OctopusToolRunner, readonly connection: OctoServerConnectionDetails) {}
+
+    public async run(
+        space: string,
+        project: string,
+        releaseNumber?: string | undefined,
+        channel?: string | undefined,
+        customReleaseNotes?: string | undefined,
+        deployToEnvironments: string[] = [],
```
[the remark this answers is not in the document]
Comment (johnsimons) [reply source/tasks/CreateOctopusRelease/CreateOctopusReleaseV5/createRelease.ts:14]: @slewis74 I think we should raise an issue about it so we all have a better understanding about the intent?

File: source/tasks/Deploy/DeployV5/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { executeTask } from "../../Utils/octopusTasks";
+
+export class Deploy {
+    constructor(readonly tool: OctopusToolRunner, readonly connection: OctoServerConnectionDetails) {}
+
+    public async run(
+        space: string,
+        project: string,
+        releaseNumber: string,
+        deployToEnvironments: string[],
```
[the remark this answers is not in the document]
Comment (johnsimons) [reply source/tasks/Deploy/DeployV5/deploy.ts:12]: @slewis74 sounds reasonable, so we all understand it better can you create individual issues for this?

File: source/tasks/BuildInformation/BuildInformationV5/task.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (slewis74) [reply source/tasks/BuildInformation/BuildInformationV5/task.json:76]: Yeah, that's a fair point 👍

File: source/tasks/CreateOctopusRelease/CreateOctopusReleaseV5/createRelease.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (slewis74) [reply source/tasks/CreateOctopusRelease/CreateOctopusReleaseV5/createRelease.ts:14]: Sounds like a good idea. I'll raise one for this and one for the tenanted/untenanted deployments too.

**Comment by johnsimons** [comment]
@slewis74 it seems you are happy to merge this, can I get the PR approved, 🙇

**Review by slewis74 — APPROVED** [review]
I have been through the code, seems ok. Haven't done any specific testing.

**Pull request merged**